### PR TITLE
Ignore `<Textarea>` components in `vue/no-textarea-mustache`

### LINKED ISSUE #2023: Rule `no-textarea-mustache` cannot difference `<Textarea>` component
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** `^8.24.0`
- **eslint-plugin-vue version:** `^9.5.1`
- **Node version:** `16.17.1`
- **Operating System:** `Windows 10`

**Please show your full configuration:**

```json5
{
  extends: [
    'plugin:vue/vue3-recommended',
    // 'standard',
    // 'plugin:jsonc/recommended-with-jsonc',
    // 'plugin:yml/standard',
    // 'plugin:markdown/recommended',
    '@vue/typescript/recommended'
  ],
  env: {
    es6: true,
    browser: true,
    node: true
  },
  parser: 'vue-eslint-parser',
  parserOptions: {
    ecmaVersion: 'latest'
  },
  reportUnusedDisableDirectives: true
}
```

**What did you do?**	

```vue
<template>
  <Textarea value="Almost" :max-length="10">
    <template #count="{ value }">
      {{ `${value.length}/10` }}
    </template>
  </Textarea>
</template>
```

**What did you expect to happen?**

No error occurred.

**What actually happened?**

Get a error:

```log
error  Unexpected mustache. Use 'v-model' instead  vue/no-textarea-mustache
```

**Repository to reproduce this issue**

You can add a valid test in [here](https://github.com/vuejs/eslint-plugin-vue/blob/master/tests/lib/rules/no-textarea-mustache.js#L25):

```json5
{
  filename: 'test.vue',
  code: '<template><div><Textarea>{{text}}</Textarea></div></template>'
}
```

Then test it and it will fail.

### PR DESCRIPTION
**Pull request by qmhc**
Fix #2023

### CODE REVIEW

File: lib/rules/no-textarea-mustache.js [hunk 1]
```diff
@@ -20,13 +20,21 @@ module.exports = {
...
+    const sourceCode = context.getSourceCode()
+
     return utils.defineTemplateBodyVisitor(context, {
       /** @param {VExpressionContainer} node */
       "VElement[name='textarea'] VExpressionContainer"(node) {
         if (node.parent.type !== 'VElement') {
           return
         }
 
+        const parentStartTag = sourceCode.getText(node.parent.startTag)
```
Comment (FloEdelmann) [inline lib/rules/no-textarea-mustache.js:32]: No need for the whole `sourceCode` here. You can use `node.parent.rawName` instead. See https://astexplorer.net/#/gist/d568be011d22bbc7c364e1eb4e2de668/b7f5155c779ee53528caa1373035b4d7436d4146

```suggestion
        const parentStartTag = node.parent.rawName
```

**Review by FloEdelmann — CHANGES_REQUESTED** [review]

File: lib/rules/no-textarea-mustache.js [hunk 2]
```diff
@@ -23,7 +23,10 @@ module.exports = {
     return utils.defineTemplateBodyVisitor(context, {
       /** @param {VExpressionContainer} node */
       "VElement[name='textarea'] VExpressionContainer"(node) {
-        if (node.parent.type !== 'VElement') {
+        if (
+          node.parent.type !== 'VElement' ||
+          node.parent.rawName.startsWith('T')
```
Comment (FloEdelmann) [inline lib/rules/no-textarea-mustache.js:28]:
```suggestion
          node.parent.rawName[0] === 'T'
```

**Review by FloEdelmann — APPROVED** [review]
Looks good to me. Thanks for reporting and fixing it yourself!

I have just one suggestion, but I don't know if it's easier to read this way.

File: lib/rules/no-textarea-mustache.js (hunk 2 above)
Comment (qmhc) [reply lib/rules/no-textarea-mustache.js:28]: I think using a named method should be easier than using notations to read.

What's your starting point about this?

Comment (ota-meshi) [reply lib/rules/no-textarea-mustache.js:28]: How about rewriting the selector?

```diff
-      "VElement[name='textarea'] VExpressionContainer"(node) {
+      "VElement[rawName ='textarea'] VExpressionContainer"(node) {`
```

Comment (qmhc) [reply lib/rules/no-textarea-mustache.js:28]: LGTM~

Comment (FloEdelmann) [reply lib/rules/no-textarea-mustache.js]: Ah, even better!

**Pull request merged**